### DISCUSSION TITLE
Question titres affectés dynamiquement ?

### DISCUSSION THREAD

**Discussion by MpmMess** [MTESSDev/FRW#8] [Annonces]
Est-ce que le titre du formulaire peut être composé en entier ou en partie par une valeur affecté dynamiquement lors de l’instanciation du formulaire? 
Si non, peut-on avoir un en-tête de page (donc répété à toutes les pages) contenant les champs, ex.: « Année »  ou autre valeur?

**Comment by anisite** [comment]
@j0ber peux tu répondre SVP ?

**Comment by j0ber** [comment]
@MpmMtess pouvez-vous préciser s.v.p. ? De quel titre vous parlez? 
![FRW-TitreForm](https://user-images.githubusercontent.com/26974817/217082114-5aa3d5d7-6f5c-4750-87e6-832fe6efdab0.jpg)

**Comment by billo04** [comment]
Dans l'image exemple, il s'agit du titre de la page (Demande d'aide financière de dernier recours ou de services d'emploi).
On veut savoir si ce titre peut être fourni, en tout ou en partie, lors de l'instanciation du formulaire.

**Reply by j0ber** [reply]
En fait j'aurait dû le dire dès le départ mais les 3 éléments ne sont pas "dynamiques" aujourd'hui.

Je ne vois pas trop l'intérêt que le titre puisse contenir une portion "dynamique". Avez-vous un cas concret qui pourrait nous inciter à revoir cette position (un exemple concret)?

**Comment by billo04** [comment]
L'organisme débute son cheminement dans la Zone Entreprise où on lui présente ses demandes comme ceci:
![image](https://user-images.githubusercontent.com/124705516/217350332-1ef28460-850d-4afa-80c1-ae62c821ad3b.png)

En cliquant sur le bouton "Continuer", il est redirigé vers FRW:
![image](https://user-images.githubusercontent.com/124705516/217350717-db57e830-a6a4-41c7-a71e-9821611348b5.png)

On voudrait personnaliser le titre du formulaire pour y présenter le numéro de la demande et l'année financière visée afin que l'organisme voit qu'il saisit le formulaire correspondant à la demande qu'il a sélectionné et ce, tout au long de la saisie de son formulaire.
![image](https://user-images.githubusercontent.com/124705516/217351487-ce366e75-89de-4401-bd55-dd3a827b7cec.png)

**Reply by billo04** [reply]
Ainsi, on n'aurait pas à répéter les champs "No demande" et "Année financière" à toutes les pages du formulaire.

**Comment by billo04** [comment]
Est-ce que FRW permet actuellement d'avoir un sous-titre de formulaire ou une en-tête de page, bref d'avoir une zone entre le titre du formulaire (ex: Programme de soutien financier des Corporations de développement communautaire) et le titre le la page (ex: Renseignements sur l'organisme | Information sur l'organisme)?

**Comment by j0ber** [comment]
Nous allons ajouter une notion de sous-titre. Ce dernier pourra être dynamique. Nous allons ajouter cela à notre backlog.

Ça va ressembler à ceci tel que convenu avec l'équipe UX : 
![image](https://user-images.githubusercontent.com/26974817/219117541-9dbb136b-da59-44b2-b73a-fe4279bd3d54.png)

**Comment by billo04** [comment]
L'emplacement de ce sous-titre laisse croire qu'il ne pourra pas être plus large que la section de gauche présentant les sections et pages du formulaire. Si tel est le cas, cette solution ne répondra pas à notre besoin.
Idéalement, le sous-titre devrait pouvoir prendre toute la largeur du formulaire et être positionné sous la barre orange du titre.

Dans un premier temps, nous souhaitons remplir ce sous-titre avec le numéro de la demande ainsi que l'année financière ciblé (tel qu'illustré ci-haut), mais ultimement, nous voudrions y mettre également le nom du programme de soutien financier. 
Des programme que nous supportons pour le moment, le plus long nom fait 160 caractères. Les futures programmes pourraient avec des noms encore plus long... prévoir suffisamment d'espace.
Nous souhaitons également pouvoir gérer le changement de ligne (injection d'un CRLF) du sous-titre afin que l'information qui y soit alignée adéquatement sans égard à la largeur de l'écran.
Ainsi, sur la première ligne on aura le nom du programme et sur la seconde, le numéro de la demande et l'année financière (voir l'image suivante).

![image](https://user-images.githubusercontent.com/124705516/230431893-3cba782a-6c8d-4a5b-af65-14601a15504a.png)

**Comment by LarcM7** [comment]
#20

**Reply by MpmMess** [reply]
Bonjour, pouvons nous avoir des exemples du code afin de pouvoir inclure la solution dans nos formulaires ?

**Reply by LarcM7** [reply]
Bonjour! 

J'ai fermé la présente discussion avec la référence à l'issue 20 afin de poursuivre le suivi à l'intérieur de l'issue #20. 

La solution n'est pas prête à être utiliser dès maintenant, nous venons tout juste de démarrer le développement.

Restez à l'affut de l'issue 20 afin d'être informé des suites pour ce besoin, des mises à jour y apparaîtront sous peu.

Nous nous assurerons de fournir toute documentation pertinente incluant des exemples afin d'accompagner les utilisateurs intéressés. 

Merci
